webpack/assets/javascripts/react_app/components/notifications/toast/Toast.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import Icon from '../../common/Icon';
+import helpers from '../../../common/helpers';
+import { ALERT_CSS } from '../../../constants';
+import CloseButton from '../../common/CloseButton';
+import * as ToastActions from '../../../redux/actions/toasts';
+// eslint-disable-next-line no-unused-vars
+import Timer from '../../common/Timer';
+
+const timerDelay = 8000;
+
+class Toast extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['dismiss', 'execute', 'onMouseEnter', 'onMouseLeave']);
+  }
+
+  componentDidMount() {
+    if (!this.props.sticky) {
+      this.timer = new Timer(this.execute, timerDelay);
+      this.timer.startTimer();
+    }
+  }
+
+  componentWillUnmount() {
```

Comment:
does this also takes into account when I'm navigating to a new page? e.g. on the login page, if i submit incorrect password, i want the toast to be removed once i submitted a new pw (and before the server response)